`head` broken when only first row-slice is needed to satisfy the range, and a dynamic string column contains `None`s or `NaN`s

Minimal repro (also broken in same way when replacing `np.nan` by `None`):
```
def test_head_with_nans(lmdb_version_store):
    lib = lmdb_version_store
    sym = "test_head_with_nans"
    df = DataFrame({"col0": ["hello", np.nan, "bonjour"]})
    lib.write(sym, df)
    lib.head(sym, n=2).data
```
Exception:
```
    def _read_dataframe(self, symbol, version_query, read_query, read_options):
>       return ReadResult(*self.version_store.read_dataframe_version(symbol, version_query, read_query, read_options))
E       arcticdb_ext.exceptions.InternalException: std::invalid_argument(Cursor overflow in chunked_buffer ptr_cast, cannot read 8 bytes from a buffer of size 26416 with cursor at 9223372036854775806, as it would required 9223372036854775814 bytes. )

../../../current/lib/python3.6/site-packages/arcticdb/version_store/_store.py:1586: InternalException
```
Caused by this commit: https://github.com/man-group/ArcticDB/commit/8643986ff55cb20a9108e01830e765cb99faa0d8
Almost certainly the hacky clause used to reduce memory usage of `head` when only a single row-slice is required is the cause, as using a small `segment_row_size` and then requesting a larger number of rows works.

--- a/cpp/arcticdb/column_store/memory_segment_impl.hpp
+++ b/cpp/arcticdb/column_store/memory_segment_impl.hpp
@@ -922,6 +922,9 @@ public:
         // Map from offsets in the input stringpool to offsets in the output stringpool
         // Only used if filter_down_stringpool is true
         emilib::HashMap<StringPool::offset_t, StringPool::offset_t> input_to_output_offsets;
+        // Prepopulate with None and NaN placeholder values to avoid an if statement in a tight loop later
+        input_to_output_offsets.insert(not_a_string(), not_a_string());
+        input_to_output_offsets.insert(nan_placeholder(), nan_placeholder());
 
         // Index is built to make rank queries faster
         std::unique_ptr<util::BitIndex> filter_idx;

